client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,461 @@
+	}
+	return timeutil.Human(t)
+}
+
+func printVerboseAssertionHeaders(w *tabwriter.Writer, options PrintModelAssertionOptions, assertion asserts.Assertion) error {
+	allHeadersMap := assertion.Headers()
+	for _, headerName := range niceOrdering {
+		invalidTypeErr := fmt.Errorf("invalid type for %q header", headerName)
+
+		headerValue, ok := allHeadersMap[headerName]
+		// make sure the header is in the map
+		if !ok {
+			continue
+		}
+
+		// switch on which header it is to handle some special cases
+		switch headerName {
+		// list of scalars
+		case "required-snaps", "system-user-authority":
+			headerIfaceList, ok := headerValue.([]interface{})
+			if !ok {
+				// system-user-authority can also appear as string
+				headerString, ok := headerValue.(string)
+				if ok {
+					fmt.Fprintf(w, "%s:\t%s\n", headerName, headerString)
+					continue
+				}
+				return invalidTypeErr
+			}
+			if len(headerIfaceList) == 0 {
+				continue
+			}
+
+			fmt.Fprintf(w, "%s:\t\n", headerName)
+			for _, elem := range headerIfaceList {
+				headerStringElem, ok := elem.(string)
+				if !ok {
+					return invalidTypeErr
+				}
+				// note we don't wrap these, since for now this is
+				// specifically just required-snaps and so all of these
+				// will be snap names which are required to be short
+				fmt.Fprintf(w, "  - %s\n", headerStringElem)
+			}
+
+		// timestamp needs to be formatted with fmtTime from the timeMixin
+		case "timestamp":
+			timestamp, ok := headerValue.(string)
+			if !ok {
+				return invalidTypeErr
+			}
+
+			// parse the time string as RFC3339, which is what the format is
+			// always in for assertions
+			t, err := time.Parse(time.RFC3339, timestamp)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "timestamp:\t%s\n", fmtTime(t, options.AbsTime))
+
+		// long string key we don't want to rewrap but can safely handle
+		// on "reasonable" width terminals
+		case "device-key-sha3-384":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return invalidTypeErr
+			}
+
+			switch {
+			case options.TermWidth > 86:
+				fmt.Fprintf(w, "device-key-sha3-384: %s\n", headerString)
+			case options.TermWidth <= 86 && options.TermWidth > 66:
+				fmt.Fprintln(w, "device-key-sha3-384: |")
+				strutil.WordWrapPadded(w, []rune(headerString), "  ", options.TermWidth)
+			}
+		case "snaps":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			snapsHeader, ok := headerValue.([]interface{})
+			if !ok {
+				return invalidTypeErr
+			}
+			if len(snapsHeader) == 0 {
+				// unexpected why this is an empty list, but just ignore for
+				// now
+				continue
+			}
+			fmt.Fprintf(w, "snaps:\n")
+			for _, sn := range snapsHeader {
+				snMap, ok := sn.(map[string]interface{})
+				if !ok {
+					return invalidTypeErr
```

Comment:
It might be helpful if all the errors returned in this `for` loop also mentioned the snap name.

Reply:
I have added this now